### ISSUE TITLE
Research IPRP tables to determine if sane pricing is possible

### ISSUE THREAD

**Issue by AgonyAunt** [AgonyAunt/Sigil-City-of-Doors-PW#139]
As part of the economic overhaul one option is to zero the cost multiplers that exist on item properties in the 2das.

Is it possible to make these values work sanely?

From my limited testing i believe it is not possible, but i would be happy to proven wrong.

Main issue: When item properties get added, they seem to multiply together.

eg: Full Plate Base cost might be 1500.
Full Plate +5 might cost 10,000. (minus base cost 8500)
Full Plate with Resist Fire 10/- might cost. 3000. (minus base cost 1500)

So we would expect Full Plate +5 with Resist Fire 10/- to cost somewhere in the region of 13000. Instead it is likely to cost something like 30000.

Can someone investigate this further?

**Comment by AgonyAunt** [comment]
@Ceremorph - is this something you could take a look at?

**Comment by Ceremorph** [comment]
I apologize for the messiness of the following, but I'm writing this stream of consciousness as I delve.

You're totally missing the correct costs in your example. Yes, Full plate is 1500, but +5 Full Plate is 25,510 (enchant value of 24,010). Full Plate with resist fire 10/ runs 4562 (enchant value of 3062). 

Now yes, here's where things get funky. Both on the same item  gives us 45,722... a considerable jump over what should be, if added together, 28,572. So it's pretty obvious that there's some sort of multiplier in place. It gets even further out of control when I add Str +1, which is only worth 1102 on its own, but jumps the price up to 60,789. So it seems rather obvious that when adding multiple enchantments, the price seems to include a multiplier of some sort.

So I dig down a bit further, to iprp_resistcost.2da. On the line describing a Resist 10/, there's a 3.5 value in the "cost" column... but I'm not sure what that applies to, because there's no actual math that makes the total price jump that way.

So let's try something different. The iprp_feats.2da also has a cost column. So, taking our base full plate and giving it Cleave, which has a value of 1, raises its value by 1000. Giving it Dodge, also a value of 1, raises its value by the same. Two-Weapon Fighting has a value of 1.5 and raises it by 2250, yet Mobility has a value of 5 and raises it by 2250.

And suddenly things get weird. Cleave and Dodge together?  The value increase by 4000.

Now here's a weird thing. In baseitems.2da, there's a column called "ItemMultiplier". This has a value of 1 for armor and 2 for longsword. When I give the cleave feat to the longsword, it does get its value raised by 2000, or double the same on armor. So it seems this is part of the calculation; the same enchantment on a weapon has double the intrinsic value of on armor... the +5 armor bonus jumps it up  48020, again double the amount. So it seems we have a starting point.

Oh shit... I think I finally found it. gonna make a new comment for what I just discovered....

**Comment by Ceremorph** [comment]
In the text file explaining itempropdef.pdf, there's the following bit of information.

"The actual calculation may or may not involve summing all the "cost" values of item properties, squaring this value, multiplying by 1000, adding this to the base value of the item, then squaring the sum of all the negative "cost" values, subtracting this from the positive/beneficial sum, then multiplying this value by the base item multiplier."

The squaring is what gets you. I think there might be a little bit of tweaking going on at the end, but this actually comes somewhat close to fitting the math and explaining why these big jumps occur. And thus, armor with +5AC and cleave... add together the 5 of the AC and the 1 of cleave and you get 6. 6^2 X 1000 = 36000... and the armor carries a value in toolset of 36310. That also fits with the earlier cleave + dodge armor, giving us (1+1)^2 X 1000, or 4000.

What this means is if we are working with small values (+1 AC has a cost value of 1, for instance), there's a small exponential boost, because you're dealing with squaring small numbers. However, when you throw a more powerful property into the mix, the jump between five and six squared is considerable.

Worst news? All of this is handled by an individual table for each property type, and even multiple tables used for some (for instance, castable spells pull their cost from both what spell is cast (it looks like the cost is the spell level) and on how many charges it spends per cast or uses per day it has. So you've got an absolute ass-ton or work in front of you if you intend to rework how much things cost...

**Comment by AgonyAunt** [comment]
Wow, thanks for the research Cere, its most useful.  Particularly love the comment from the PDF about "may or may not".... typical.

And yes, this is why i thought the best way to go many be to simply zero all those cost values and set item values manually. A lot of work for sure, but less of a pain than trying to find sane values for items.

**Comment by AgonyAunt** [comment]
Any reason you reopened?

**Comment by Ceremorph** [comment]
Figured you as the topic starter would want to close it yourself, particularly if you were going to put another comment here or ask a follow-up question.

Oh, and I did realize a possible secondary methodology of doing this. Right now, things are pretty much across the board so far as that cost number goes. I think one possible workaround, which would be easier than manually figuring the potential value of something, would be to simplify the system. The idea of zeroing them all... this would cause enchanted items to not gain in value. Instead, assuming there will still be some sort of cap on sell values, we could get rid of some of the "OMG this is worth two million" items by simply going with a lower-tiered system, with whole numbers and a maximum value of 3 (thus low, medium and high value for every enchant). In addition, we could make every enchanted/crafted object type have a base value of 1 (instead of as now, where weapons have a double base enchant value compared to armor). This would result in total values ranging between +1000 for simple enchants to a maximum of 81,000 (9^2) for a three-enchant item with all "highs" and no spellcasting capabilities. In the same way we could simplify how spellcasting items grow in value as well.

The one fortunate thing is that all of the 2das that are used to calculate these things are easy to find, and if we were just simplifying without manually rating every single possible boost, instead using the existing values as a baseline (1 staying 1, 2 and 3 becoming 2, and above 3 becoming 3). So... yeah, probably a quick dose of Excel could mass-edit each of the different files with a well-constructed IF function.,, and with the spells, this could be done at the same time as setting up the different levels if we wanted to go by that method.
